cordformation/src/main/kotlin/net/corda/plugins/Dockerform.kt

```diff
@@ -201,6 +212,29 @@ open class Dockerform @Inject constructor(objects: ObjectFactory) : Baseform(obj
             services[it.containerName] = service
         }
 
+        if (external.containerImage.isPresent()) {
+
+            val extra = mutableMapOf(
+                    "container_name" to external.containerName.get(),
+                    "image" to external.containerImage.get(),
+                    "ports" to external.servicePorts.get().map {QuotedString("${it}:${it}") },
+                    "expose" to external.servicePorts.get()
+            )
+            if(external.volumes.get().isNotEmpty()){
+                extra["volumes"] = external.volumes.get().map{ "${it["sourceFile"]}:${it["deploymentPath"]}" }
+            }
+            if(external.environment.get().isNotEmpty()){
+                extra["environment"] = external.environment.get().map { "${it.key}=${it.value}" }
+            }
+            if(external.privileged.get() == true) {
+                extra["privileged"] = external.privileged.get()
+            }
+            if(external.commands.get().isNotEmpty()){
```

Comment:
I would remove the convention from `commands` too, and then say
```kotlin
if (external.commands.isPresent()) {
    extra["command"] = external.commands.get()
}
```
And then annotate `commands` as `@Optional`, of course.